### ISSUE TITLE
Training plataueing after just 35 epochs.

### ISSUE THREAD

**Issue by VeryFastSnail** [OHF-Voice/piper1-gpl#75]
I have tried to train Lithuanian model using default parameters from training docs and I'm facing plateau for eval_loss after just 34 epochs. lightning logs show a good progress and audio sounds acceptable, but still a bit robotic. I have implemented stopping on plateauing with simple patience of 20.   

My dataset is 34k sentences consisting of single speaker (female), text is normalized and cleaned up as good as I can get it. Audio recorded in controlled environment and with good quality microphone, no external noises.

What could be the cause for this? Should I let it run for 300 epochs and just extract best checkpoint after that? Could Learning rate scheduler be implemented? stochastic_weight_avg.Stocha?  is it possible on piper training script? 

I use rtx 3090 62 batches and 16-mixed precision, I tried evaluating every epoch, every 5, same result

If I add different speaker with 2000-2500 sentences, could it improve the sound and emotion of the voice?

This is my latest config:

```
seed_everything: 1234

data:
  voice_name: "Aira"
  csv_path: "{redacted}/metadata_normalized.csv"
  audio_dir: "{redacted}/normalized_wavs"
  cache_dir: "{redacted}/.cache"
  espeak_voice: "lt"
  config_path: "{redacted}/model.onnx.json"
  batch_size: 62
  num_workers: 23

model: {}

trainer:
  accelerator: gpu
  devices: 1
  precision: "16-mixed"
  max_epochs: 3000
  check_val_every_n_epoch: 1
  val_check_interval: 1.0
  log_every_n_steps: 50
  enable_checkpointing: true
  callbacks:
    - class_path: lightning.pytorch.callbacks.EarlyStopping
      init_args:
        monitor: val_loss
        mode: min
        min_delta: 0.001
        patience: 80
        verbose: true

    - class_path: lightning.pytorch.callbacks.ModelCheckpoint
      init_args:
        monitor: val_loss
        mode: min
        save_top_k: 5
        save_last: true
        filename: "{epoch:03d}-{val_loss:.4f}"

    - class_path: lightning.pytorch.callbacks.LearningRateMonitor
      init_args:
        logging_interval: epoch

    #disabled,since idk if piper allows it
    # - class_path: lightning.pytorch.callbacks.stochastic_weight_avg.StochasticWeightAveraging
    #   init_args:
    #     swa_lrs: 1.0e-4
    #     swa_epoch_start: 0.9

```

**Comment by synesthesiam** [comment]
I trained on the entire LJSpeech dataset, which was around 24 hours worth of audio.

Depending on your dataset's license, I'd be happy to help train a model.

**Comment by FatihcanUslu** [comment]
@VeryFastSnail thanks — here are the details:

- On average my .wav files are ~10 seconds long, total dataset ~6 hours.
- This is the command I run:

'''
python3 -m piper.train fit \
  --data.voice_name "youtube" \
  --data.csv_path {redacted}/piper/dataset_youtube/metadata.csv \
  --data.audio_dir {redacted}/piper/dataset_youtube/wavs \
  --model.sample_rate 22050 \
  --data.espeak_voice "en" \
  --data.cache_dir {redacted}/piper/piper1-gpl/outputs/cache \
  --data.config_path {redacted}/piper/piper1-gpl/outputs/finetuned_model/config.json \
  --data.batch_size 6 \
  --ckpt_path /tmp/tmpeks7hlac.ckpt \
  --data.num_workers 2 \
  --trainer.accelerator gpu \
  --trainer.precision 16-mixed \
  --trainer.enable_checkpointing true
'''
When I train on the whole dataset with batch_size=6, GPU memory slowly climbs up to ~23GB and stays there.  If I restrict the dataset to only ~10 minutes of audio (a small subset with similar average clip length ~10 seconds), peak usage is ~7GB. To me, the memory usage should be roughly the same.
